tasks/giovanni-adapter/app/stac/catalog.ts

```diff
@@ -0,0 +1,69 @@
+import * as fs from 'fs';
+import { strict as assert } from 'assert';
+import path from 'path';
+import { v4 as uuid } from 'uuid';
+import _ from 'lodash';
+import StacItem from './item';
+import { StacCatalog, StacLink } from './types';
+
+/**
+ * Implementation of the StacCatalog type with constructor, write function, and ability
+ * to add children
+ */
+export default class Catalog implements StacCatalog {
```

Comment:
It feels like we should just define the Stac data structures and functions in a single library and use them throughout. You don't need to do that for this PR, but we should have a ticket for it.

Reply:
Good point ... story created:
https://bugs.earthdata.nasa.gov/browse/HARMONY-1101